test/e2e/autonomy/autonomy_test.go

```diff
@@ -0,0 +1,330 @@
+				kubeProxyContainerID = strings.Trim(string(opBytes), "\r\n")
+				return true
+			}).Should(gomega.BeTrue(), "fail to get kube-proxy container ID")
+			// restart kube-proxy
+			gomega.Eventually(func() bool {
+				_, err := exec.Command("/bin/bash", "-c", "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'crictl stop "+kubeProxyContainerID+"'").CombinedOutput()
+				if err != nil {
+					return false
+				}
+				return true
+			}).Should(gomega.BeTrue(), "fail to stop kube-proxy")
+
+			gomega.Eventually(func() bool {
+				_, err := exec.Command("/bin/bash", "-c", "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'iptables -F'").CombinedOutput()
+				if err != nil {
+					return false
+				}
+				return true
+			}).Should(gomega.BeTrue(), "fail to remove iptables on node "+YurtEdgeNode2Name)
+
+			gomega.Eventually(func() string {
+				opBytes, err := exec.Command("/bin/bash", "-c", "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'curl "+NginxServiceIP+"'").CombinedOutput()
+				if err != nil {
+					return ""
+				}
+				return string(opBytes)
+			}).WithTimeout(10*time.Second).WithPolling(1*time.Second).Should(gomega.ContainSubstring("nginx"), "fail to read curl response from service: "+NginxServiceName)
+		})
+	})
+
+	var _ = ginkgo.Describe("coredns"+YurtE2ENamespaceName, func() {
+		ginkgo.It("coredns edge-autonomy test", ginkgo.Label("edge-autonomy"), func() {
+			ginkgo.Skip("current coredns does not support edge-autonomy, coredns-edge-autonomy tests will be skipped.")
+			// obtain coredns containerID with crictl
+			gomega.Eventually(func() bool {
+				cmd := ` /bin/bash -c "crictl ps | grep coredns | awk '{print \$1}'"`
+				opBytes, err := exec.Command("/bin/bash", "-c", "docker exec -t "+coreDnsNodeName+cmd).CombinedOutput()
+				if err != nil {
+					return false
+				}
+				coreDnsContainerID = strings.Trim(string(opBytes), "\r\n")
+				return true
+			}).Should(gomega.BeTrue(), "fail to get coredns container ID")
+			// restart kube-proxy
+			gomega.Eventually(func() bool {
+				cmd := "docker exec -t " + coreDnsNodeName + " /bin/bash -c 'crictl stop " + coreDnsContainerID + "'"
+				_, err := exec.Command("/bin/bash", "-c", cmd).CombinedOutput()
+				if err != nil {
+					return false
+				}
+				return true
+			}).Should(gomega.BeTrue(), "fail to stop coredns")
+			gomega.Eventually(func() string {
+				cmd := "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'dig " + "@" + CoreDNSServiceIP + " " + NginxServiceName + "." + YurtDefaultNamespaceName + ".svc.cluster.local'"
+				opBytes, err := exec.Command("/bin/bash", "-c", cmd).CombinedOutput()
+				if err != nil {
+					return ""
+				}
+				return string(opBytes)
+			}).WithTimeout(10*time.Second).WithPolling(1*time.Second).Should(gomega.ContainSubstring("NOERROR"), "DNS resolution contains error, coreDNS dig failed")
+		})
+	})
+})
+
+func TestEdgeAutonomy(t *testing.T) {
+	ginkgo.BeforeSuite(func() {
+		gomega.RegisterFailHandler(ginkgowrapper.Fail)
```

Comment:
Do not put `BeforeSuite and AfterSuite` into Test function, and please move `RegisterFailHandler` into Test function.
We should have the format:
```go
var _ = BeforeSuite(func() {

})
var _ = AfterSuite(func() {
})
func Testxxx(t *tesing.T) {
  gomega.RegisterFailHandler
  ginkgo.RunSpecs
}
```

Reply:
will do